### DISCUSSION TITLE
geckodriver needs profile for gauge testing

### DISCUSSION THREAD

**Discussion by ademant** [getgauge/gauge#2216] [General]
Hi,
the testing with chrome (chromedriver) works really good. But I can't get testing with geckodriver working (with Javascript). Several internet sites give hints about using, but none of them are working:
- https://www.selenium.dev/documentation/
- https://www.selenium.dev/selenium/docs/api/javascript/
- https://www.codementor.io/@nehavaidya/what-are-chromedriver-and-geckodriver-in-selenium-uijax855e
- https://stackoverflow.com/questions/58266549/connect-to-geckodriver-with-selenium-js

Best thing is: Geckodriver is working, but it complain about missing profile, even if I give the directory to a working profile.

Using:
Ubuntu Impish
Gauge 1.4.3
- js 2.4.0
- html-report 4.1.4
- geckodriver 0.30.0

**Comment by zabil** [comment]
This might not be related to Gauge, is there a sample project you can share (preferably as a github repo) to replicate this?

**Reply by ademant** [reply]
No, I tried the JS-Code in the above examples and none worked.
Maybe it is not strict a topic related to Gauge; but Gauge advertise itself to be able to use different browser. And then Gauge explain only usage of chromedriver...

**Reply by zabil** [reply]
> but Gauge advertise itself to be able to use different browser. 

Gauge actually does not have that capability. It relies on an underlying driver. Sorry about the wording if that's confusing, can you point out where there's a mention about it working with different web browsers? 

In the meanwhile maybe more details about the exact problem (a sample project) with log messages etc might help someone who's got it working debug the issue faster.

**Reply by ademant** [reply]
Maybe I'm too frustrated about Gauge, so I mix information from different sources. Maybe Gauge itself does not advertise to be able to use different browser.
But there are so many things, not working and no information how to solve it for gauge, I'll look for alternatives.
For example: Few weeks ago you opened an issue out of my question, but since then nothing happened.

**Reply by zabil** [reply]
Sure, sorry to hear that and thanks for the feedback. 

Perhaps, it's not clear Gauge (and Taiko) websites and forums that it's currently maintained on spare time of the original contributors and is not that active anymore. Will be posting something about that soon.

>But there are so many things, not working and no information how to solve it for gauge, I'll look for alternatives.

Agree. Please do look at alternatives. Thanks again.
